### PR TITLE
Fix incorrect id on search/searching on non-hex values

### PR DESCRIPTION
**Pull request by shjortConcordium**
## Purpose

Fix incorrect Id appearing on tokens when searching.

## Changes

Perform more check before searching for tokens

### CODE REVIEW

File: packages/browser-wallet/src/popup/pages/Account/ManageTokens/TokenList.tsx [hunk 1]
```diff
@@ -84,6 +84,10 @@ const InfiniteTokenList = forwardRef<HTMLDivElement, InfiniteTokenListProps>(
     }
 );
 
+function isEven(i: number) {
+    return i % 2 === 0;
+}
```
Comment (soerenbf) [inline packages/browser-wallet/src/popup/pages/Account/ManageTokens/TokenList.tsx:89]: Wait.. I thought `isHex` already did this? Didn't we discuss this at some point?

File: packages/browser-wallet/src/popup/pages/Account/ManageTokens/TokenList.tsx [hunk 2]
```diff
@@ -105,7 +109,7 @@ const lookupTokenIdConfigure = (
 
             let value: ContractTokenDetails[] = [];
 
-            if (!searchQuery) {
+            if (!searchQuery || !isHex(searchQuery) || !isEven(searchQuery.length)) {
```
Comment (soerenbf) [inline packages/browser-wallet/src/popup/pages/Account/ManageTokens/TokenList.tsx:112]:
```suggestion
            if (!searchQuery || !isHex(searchQuery) || !isEven(searchQuery.length)) {
                setResult({ loading: false, value: undefined });
```

And then remove the conditional at line 179-181 (which does exactly the same, just prior to entering this function).

**Review by soerenbf — CHANGES_REQUESTED** [review]

File: packages/browser-wallet/src/popup/pages/Account/ManageTokens/TokenList.tsx (hunk 1 above)
Comment (shjortConcordium) [reply packages/browser-wallet/src/popup/pages/Account/ManageTokens/TokenList.tsx:89]: It doesn't seem to. :shrug:

File: packages/browser-wallet/src/popup/pages/Account/ManageTokens/TokenList.tsx (hunk 2 above)
Comment (shjortConcordium) [reply packages/browser-wallet/src/popup/pages/Account/ManageTokens/TokenList.tsx:112]: :thinking: 
Why?

Comment (soerenbf) [reply packages/browser-wallet/src/popup/pages/Account/ManageTokens/TokenList.tsx:112]: The conditional at 179-181 only runs on `!searchQuery`, but the one marked above is more lax. As such, you could end in a place where the latest `setResult` call is `{loading: true, ...}`.

Either the 2 conditionals need to be aligned, or they should be consolidated as requested in my previous comment.

File: packages/browser-wallet/src/popup/pages/Account/ManageTokens/TokenList.tsx (hunk 1 above)
Comment (shjortConcordium) [reply packages/browser-wallet/src/popup/pages/Account/ManageTokens/TokenList.tsx:89]: It doesn't: (The one that was referenced)
https://github.com/Concordium/concordium-node-sdk-js/blob/64baa0938fbddd813b68fdd91b012769667fe16c/packages/common/src/util.ts#L96

**Review by soerenbf — APPROVED** [review]

**Pull request merged**